sympy/polys/polytools.py

```diff
@@ -5961,7 +5961,7 @@ def _symbolic_factor(expr, opt, method):
     if isinstance(expr, Expr) and not expr.is_Relational:
         if hasattr(expr,'_eval_factor'):
             return expr._eval_factor()
-        coeff, factors = _symbolic_factor_list(together(expr), opt, method)
+        coeff, factors = _symbolic_factor_list(together(expr, fraction=opt.get('fraction', True)), opt, method)
```

Comment:
@jksuom  still, there is a problem, maybe because `fraction` gets a value `None` during different calls. It should remain the actual value (that is passed) in all calls. Shouldn't it?

Reply:
`fraction` can be handled in `_generic_factor` and added there to `opt` as follows:
```
def _generic_factor(expr, gens, args, method):
    """Helper function for :func:`sqf` and :func:`factor`. """
    fraction = args.pop('fraction', True)
    options.allowed_flags(args, [])
    opt = options.build_options(gens, args)
    opt['fraction'] = fraction
    return _symbolic_factor(sympify(expr), opt, method)
```

Reply:
This could probably be `fraction=opt['fraction']`.

Reply:
But we actually want the other way round (`opt` should contain `fraction` value).

Reply:
It should now contain the value. Therefore it should not be necessary to use `.get` with default. What do you mean by " the other way round"?

Reply:
Oh sorry! I miss understood your comment. I thought you were referring your code snippet :smile: